cmd/status.go

```diff
@@ -46,6 +46,7 @@ func statusCommand(cmd *cobra.Command, args []string) {
 			continue
 		}
 		utils.InDirectory(module.Path, func() {
+			commitId := git.GitValue("rev-parse", "HEAD")
```

Comment:
How about using the `--short` option here to keep the output more readable?

Reply:
Yeah, I wasn't sure, because the short ids might become ambiguous over time. But I guess that's not a big deal

Reply:
@mpfz0r The command should take care of that:

> Same as --verify but shortens the object name to a unique prefix with at least length characters.

Reply:
@bernd How so? Aren't they always as short as possible? But over time, with more commits the hashes need to become longer to avoid collisions.

Reply:
Should be solvable  https://stackoverflow.com/a/27428930

Reply:
> @bernd How so? Aren't they always as short as possible? But over time, with more commits the hashes need to become longer to avoid collisions.

@mpfz0r Since Git knows about every commit ID it can ensure the uniqueness of the shortened ID. If I run this on my project I can see different lengths already:

```
$ gpc status
Current project status
  Manifests: [manifests/master.json]
  Module versions
    graylog-parent                          4.3.0-SNAPSHOT (commit: c27cb8fe75, branch: master)
    graylog-plugin-collector                4.3.0-SNAPSHOT (commit: 6d34e70, branch: master, parent: 4.3.0-SNAPSHOT)
    graylog-plugin-aws                      4.3.0-SNAPSHOT (commit: 3203a3e, branch: master, parent: 4.3.0-SNAPSHOT)
    graylog-plugin-threatintel              4.3.0-SNAPSHOT (commit: 8c1962e, branch: master, parent: 4.3.0-SNAPSHOT)
    graylog-plugin-enterprise-parent        4.3.0-SNAPSHOT (commit: eb1fcf95, branch: master, parent: 4.3.0-SNAPSHOT)
    graylog-plugin-enterprise-integrations  4.3.0-SNAPSHOT (commit: 524411f, branch: master, parent: 4.3.0-SNAPSHOT)
    graylog-plugin-integrations             4.3.0-SNAPSHOT (commit: 334a301, branch: master, parent: 4.3.0-SNAPSHOT)
```

Reply:
@bernd Sure, this works _now_ but the commit ID might be ambiguous in the _future_

Reply:
> @bernd Sure, this works _now_ but the commit ID might be ambiguous in the _future_

Ah, okay. But why would that be important? This should only be relevant if this output it stored somewhere.

Reply:
@bernd Jenkins keeps all the build logs. And hypothetically we might want to reconstruct what happened on a year old build.  But as I said, it's fine this way. I just wanted to explain my reasoning for not using --short :)
please merge :+1:

Reply:
@mpfz0r Alright, got it. I was missing that context, sorry.

We can do the following to have full commit IDs in the build logs:

```diff
diff --git cmd/status.go cmd/status.go
index 1c61445..8c61a7a 100644
--- cmd/status.go
+++ cmd/status.go
@@ -3,7 +3,7 @@ package cmd
 import (
 	"strconv"
 
-	"github.com/Graylog2/graylog-project-cli/config"
+	c "github.com/Graylog2/graylog-project-cli/config"
 	"github.com/Graylog2/graylog-project-cli/git"
 	"github.com/Graylog2/graylog-project-cli/logger"
 	"github.com/Graylog2/graylog-project-cli/manifest"
@@ -29,7 +29,7 @@ func init() {
 }
 
 func statusCommand(cmd *cobra.Command, args []string) {
-	config := config.Get()
+	config := c.Get()
 	manifestFiles := manifest.ReadState().Files()
 	project := p.New(config, manifestFiles)
 
@@ -46,7 +46,14 @@ func statusCommand(cmd *cobra.Command, args []string) {
 			continue
 		}
 		utils.InDirectory(module.Path, func() {
-			commitId := git.GitValue("rev-parse", "--short", "HEAD")
+			var commitId string
+			if c.IsCI() {
+				// Don't display shortened commit IDs in CI environments to avoid future ambiguities regarding the
+				// shortened commit IDs
+				commitId = git.GitValue("rev-parse", "HEAD")
+			} else  {
+				commitId = git.GitValue("rev-parse", "--short", "HEAD")
+			}
 			revision := git.GitValue("rev-parse", "--abbrev-ref", "HEAD")
 			gitStatus := git.GitValue("status", "--porcelain")
 			filesModified, filesDeleted, filesAdded := 0, 0, 0

```

Reply:
I do not disagree with any conclusions. But I do want to point out that Jenkins build logs are deleted after 90 days.

https://github.com/Graylog2/ci-pipeline-shared/blob/master/vars/buildSnapshot.groovy#L80

Reply:
Cool :+1: Please add

Reply:
> I do not disagree with any conclusions. But I do want to point out that Jenkins build logs are deleted after 90 days.
> 
> https://github.com/Graylog2/ci-pipeline-shared/blob/master/vars/buildSnapshot.groovy#L80

@malcyon Thanks for the heads-up. I think it's still good to print the full commit IDs for that use case. A collision could show up every time.